app/src/lib/components/PowerTable.svelte

```diff
@@ -725,10 +725,11 @@ function handleSubmittedEdits(event: CustomEvent){
     let rowIndex: number = event.detail.rowIndex;
     let row = data[pageData[rowIndex][dataIdKey]];
     let rowInputs = (<HTMLInputElement>event.detail.domEvent.target).closest('tr')?.querySelectorAll(`[data-name=edit-input]`);
-    
-    if (!rowInputs) {
-        // ToDo: error should be easily visible to the end user
-        console.log('error');
+
+    const numRenderedColumns = instructs.length - 1 // minus one for checkbox column
+    if (rowInputs.length !== numRenderedColumns) {
```

Comment:
This check means if a user wants to make a column non-editable, their "constant" component would still need to use the `data-name=edit-input` to not cause an error:
```svelte
<!-- my non-editable component -->
<script>export let instructKey;export let instructTitle;export let value;export let rowIndex;export let rowId;</script>
<div data-name=edit-input>
	<p>{value}</p>
</div>
```

Reply:
This needs to be tested. I haven't tried it but it seems like the cell with `<div data-name=edit-input>` would bypass the check but it's corresponding value will get replaced by an empty string. Also having `data-name=edit-input` for a `div` is not ideal. Additionally, there might be cases that each cell would have more than one input. For example, an RGB representation of a color with inputs for Red, Green, and Blue numbers.

If you can find a way to make all edit components in a row report their final values once any submit button on that row in clicked, we won't have to deal with the complexity of each custom edit component and we won't have to resort to `querySelectorAll` to collect the values.

Reply:
> If you can find a way to make all edit components in a row report their final values once any submit button on that row in clicked, we won't have to deal with the complexity of each custom edit component and we won't have to resort to `querySelectorAll` to collect the values.

Actually, it's probably better (and also easier) if we keep track of all the "staged" (unsubmitted) changes in an Svelte store! So the edit components are responsible to write changes to this store and once an edit is submitted for row x, the staged changes of the row x will override the row data. This way even if we navigate back and forth between pages, we won't lose the staged changes. Also, we would be able to keep track of which cells have changed, so if user is closing the tab or exiting the edit mode, we can warn them about unsaved changes. This doesn't need to be a part of this PR though. For now an alert would suffice.

Reply:
Fortunately the value doesn't get set to an empty string, but I didn't do very robust testing. 

> Additionally, there might be cases that each cell would have more than one input. For example, an RGB representation of a color with inputs for Red, Green, and Blue numbers.

Interesting. That seems like a use case that would come up often. The current code for writing the data back to the `row` does not account for multiple inputs per component either, so I would need to update that too.

Let me think about this some more. I was actually looking into getting rid of `querySelectorAll` when I first started. From what I saw, there isn't an easy way to accomplish, but I didn't dig too much.

Reply:
Oops, didn't see your most recent comment before responding yesterday.

Makes sense to do the Stores work in another PR. I actually did some work before seeing your comment, and got *something* working. Whether it is actually reasonable is another story. I'm meddling with powers I cannot possibly comprehend 😛 

In regards to getting multiple Inputs per edit-component working, I don't see a way to support that while relying on the `HTMLInputElement.value`. The client would need to provide one final value for us save, maybe via data attributes. I could work on that, or put effort towards the Stores approach?

Reply:
I think for now supporting only one input per cell is fine and would work for most users.

Reply:
Ok how would you like to proceed? This PR should now be enough to fix the original issue, but the Stores approach sounds like it would be better, and would end up changing a lot of things. I could start work on new branch and close this PR?

Reply:
For use of Svelte stores, I opened https://github.com/muonw/muonw-powertable/issues/34